test/smoke/src/helpers/appiumHelper.ts

```diff
@@ -0,0 +1,98 @@
+            appiumProcess.kill();
+        }
+    }
+
+    public static prepareAttachOptsForAndroidActivity(applicationPackage: string, applicationActivity: string,
+    platformVersion: string = "9", deviceName: string = "emulator-5554") {
+        return {
+            desiredCapabilities: {
+                browserName: "",
+                platformName: "Android",
+                platformVersion: platformVersion,
+                deviceName: platformVersion,
+                appActivity: applicationActivity,
+                appPackage: applicationPackage,
+                automationName: "UiAutomator2"
+            },
+            port: 4723,
+            host: "localhost",
+        };
+    }
+
+    // Check if appPackage is installed on Android device for waitTime ms
+    public static async checkAppIsInstalled(appPackage: string, waitTime: number) {
+        let awaitRetries: number = waitTime / 1000;
+        let retry = 1;
+        await new Promise((resolve, reject) => {
+            let check = setInterval(async () => {
+                if (retry % 5 === 0) {
+                    console.log(`*** Check is app installed with command 'adb shell pm list packages ${appPackage}' for ${retry} time`);
+                }
+                let result = cp.execSync(`adb shell pm list packages ${appPackage}`).toString().trim();
+                if (result) {
+                    clearInterval(check);
+                    console.log("*** Installed React Native app found, await 10s for initializing...")
+                    await new Promise(resolve => setTimeout(resolve, 10000));
+                    resolve();
+                } else {
+                    retry++;
+                    if (retry >= awaitRetries) {
+                        reject(`${appPackage} not found after ${waitTime}ms`);
+                    }
+                }
+            }, 1000);
+        });
+    }
+
+    public static webdriverAttach(attachArgs: any) {
+        // Connect to the emulator with predefined opts
+        return wdio.remote(attachArgs);
+    }
+
+    public static async enableRemoteDebugJSForRN(client: wdio.Client<void>) {
+        console.log("*** Enabling Remote JS Debugging for application...");
+        await client.init()
+        .waitUntil(async () => {
+            cp.exec("adb shell input keyevent 82");
+            await new Promise(resolve => setTimeout(resolve, 300));
+            if (client.isExisting(this.RN_ENABLE_REMOTE_DEBUGGING_BUTTON)) {
+                console.log("*** Debug JS Remotely button found...");
+                client.click(this.RN_ENABLE_REMOTE_DEBUGGING_BUTTON);
+                console.log("*** Debug JS Remotely button clicked...");
+                return true;
+            } else if (client.isExisting(this.RN_STOP_REMOTE_DEBUGGING_BUTTON)) {
+                console.log("*** Stop Remote JS Debugging button found...");
+                return true;
+            }
+            return false;
```

Comment:
Should we throw an error instead? That'd signalize in our test that something has been changed in basic application without making us to guess.

Reply:
WaitUntil retries our commands until it returns true, otherwise it reach timeout and throw.